decidim-core/lib/decidim/core.rb

```diff
@@ -97,7 +97,25 @@ def self.seed!
   config_accessor :geocoder
 
   # Exposes a configuration option: a custom method to generate references
-  config_accessor :calculate_reference_method
+  # Default: Calculates a unique reference for the model in
+  # the following format:
+  #
+  # "BCN-DPP-2017-02-6589" which in this example translates to:
+  #
+  # BCN: A setting configured at the organization to be prepended to each reference.
+  # PROP: Unique name identifier for a resource: Decidim::Proposals::Proposal (MEET for meetings or PROJ for projects).
+  # 2017-02: Year-Month of the resource creation date
+  # 6589: ID of the resource
+  config_accessor :resource_reference_generator do
+    lambda do |resource|
```

Comment:
Maybe we could pass in `feature` as a parameter in order to decouple the implementation from the structure a bit. This way we can identify `participatory_process` and `organization` without further issues.